src/MongoDB.Driver.Core/Core/Operations/RetryabilityHelper.cs

```diff
@@ -109,6 +109,10 @@ public static bool IsResumableChangeStreamException(Exception exception, Semanti
             {
                 return true;
             }
+            if (exception is MongoCursorNotFoundException)
+            {
+                return true;
+            }
```

Comment:
Can MongoCursorNotFoundException be added to the __resumableChangeStreamExceptions set?

Or does this test have to be earlier than line 133?

Reply:
It's not possible since according to specification ([link](https://github.com/mongodb/specifications/blob/master/source/change-streams/change-streams.rst#why-is-cursornotfound-special-cased-when-determining-resumability)):

> To workaround this, drivers always treat this error as resumable despite it not having the proper error label.

Which means we should ALWAYS retry `CursorNotFound` error and the line below (117) prevents it for all servers higher than 4.3.0, so this check is in the correct place.

Not done.

Reply:
OK